docs/writing-stories/typescript.md

```diff
@@ -0,0 +1,149 @@
+---
+title: 'Writing stories in TypeScript'
+---
+
+Writing your stories in [TypeScript](https://www.typescriptlang.org/) makes you more productive. You don't have to jump between files to look up component props. Your code editor will alert you about missing required props and even autocomplete prop values. Plus, Storybook infers those component types to auto-generate the [Controls](../api/doc-block-controls.md) table.
+
+Using TypeScript also makes your code more robust. When authoring stories, you're replicating how a component will be used within your app. With type checking, you can catch bugs and edge cases as you code.
+
+Storybook has built-in TypeScript support, so you can get started with zero configuration required.
+
+## Typing stories with `Meta` and `StoryObj`
+
+When writing stories, there are two aspects that are helpful to type. The first is the [component meta](./introduction.md#default-export), which describes and configures the component and its stories. In a [CSF file](..api.md/csf), this is the default export. The second is the [stories themselves](./introduction.md#defining-stories).
+
+Storybook provides utility types for each of these, named `Meta` and `StoryObj`. Here's an example CSF file using those types:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'angular/typed-csf-file.ts.mdx',
+    'web-components/typed-csf-file.ts.mdx',
+    'common/typed-csf-file.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+### Optional type parameter
```

Comment:
Can you elaborate on the need for this heading when the text within it relates to the above? Couldn't it be removed and the text below adjusted to provide the information explicitly? More even as we're teaching users to use this pattern in all the examples we provide, and adding this may add some confusion simply by referencing it as optional. 

What do you think?